v2/anomalydetection/frontend-setup.mdx

```diff
@@ -0,0 +1,102 @@
+---
+sidebar_position: 2
+title: Frontend Setup
+hide_title: true
+---
+
+# Setting up the Frontend
+
+Setting up the frontend is relatively simple. You need to generate a request ID for each authentication event attempt. 
+
+:::important
+This frontend setup is required only for bot detection and anomaly IP based detection such as impossible travel detection. We also recommend checking for bot detection only on the email password login flows. 
+:::
+
+### Generating a Request ID on the Frontend
+
+In order to generate a request ID, you need to import and initialise the SDK using your public API key. This SDK is used to generate a unique request ID for each authentication event attempt.
+
+Below is an example of how to implement request ID generation on your frontend:
+
+
+```tsx
+const API_KEY = "<api-key>"; // Your public API key that you received from the SuperTokens team
+const SDK_URL = "https://supertokens.io/JfNbWqtXUiGYuTqb/PYmK3PIPqdvHw6iz";
+
+const supertokensDeviceIdPromise = import(SDK_URL + "?apiKey=" + API_KEY).then((RequestId: any) => RequestId.load({
+    endpoint: [
+        SDK_URL,
+        RequestId.defaultEndpoint
+    ]
+}));
+
+async function getRequestId() {
+    const deviceId = await supertokensDeviceIdPromise;
+    const result = await deviceId.get();
+    return result.requestId;
+}
+```
+
+### Passing the Request ID to the Backend
+
+Once you have generated the request ID on the frontend, you need to pass it to the backend. This is done by including the `requestId` property along with the value as part of the preAPIHook body from the initialisation of the recipes.
+
+:::important
+If the request ID is not passed to the backend, the anomaly detection will only be able to detect password breaches and brute force attacks.
+:::
+
+Below is a full example of how to configure the SDK and pass the request ID to the backend:
+```tsx
+import EmailPassword from "supertokens-auth-react/recipe/emailpassword";
+
+const API_KEY = "<api-key>"; // Your public API key that you received from the SuperTokens team
+const SDK_URL = "https://supertokens.io/JfNbWqtXUiGYuTqb/PYmK3PIPqdvHw6iz";
+
+const supertokensDeviceIdPromise = import(SDK_URL + "?apiKey=" + API_KEY).then((RequestId: any) => RequestId.load({
```

Comment:
need to also add the right tags that we give to the user?

Reply:
done